extensions/variants.go

```diff
@@ -82,6 +240,74 @@ type WindowFunctionVariant struct {
 	impl        WindowFunctionImpl
 }
 
+func NewWindowFuncVariant(id ID) *WindowFunctionVariant {
+	return &WindowFunctionVariant{
+		name: id.Name,
+		uri:  id.URI,
+		impl: WindowFunctionImpl{
+			AggregateFunctionImpl: AggregateFunctionImpl{
+				Decomposable: DecomposeNone,
+			},
+			WindowType: PartitionWindow,
+		},
+	}
+}
+
+type WindowVariantOpts struct {
+	Variadic         *VariadicBehavior
+	SessionDependant bool
+	Deterministic    bool
+	Ordered          bool
+	// value of 0 == unlimited
+	MaxSet       uint
+	Decomposable DecomposeType
+	// should be a type expression
+	// must not be empty if decomposable is not DecomposeNone
+	IntermediateOutputType string
+	WindowType             WindowType
+}
+
+func NewWindowFuncVariantOpts(id ID, opts WindowVariantOpts) *WindowFunctionVariant {
+	var aggIntermediate parser.TypeExpression
+	if opts.Decomposable == "" {
```

Comment:
I don't have a strong preference, but just curious on plus/minus of using const strings versus enum with iota. It'd prevent this kind of translation, but then you'd need a String method. For example:
```
type DecomposeType int

const (
	DecomposeNone DecomposeType = iota
	DecomposeOne
	DecomposeMany
)
```
What are your thoughts?

Reply:
So I had originally done this using the const strings so that it could easily and freely get the conversion from YAML. but i think you're right that it would be better to use `iota` and add a `String` method + `UnmarshalYAML` etc. methods